projects/plugins/protect/src/class-scan-history.php

```diff
@@ -0,0 +1,383 @@
+		if ( ! $has_required_plan ) {
+			return false;
+		}
+
+		if ( self::$history !== null ) {
+			return self::$history;
+		}
+
+		if ( $refresh_from_wpcom || ! self::should_use_cache() || self::is_cache_expired() ) {
+			$history = self::fetch_from_api();
+		} else {
+			$history = self::get_from_options();
+		}
+
+		if ( is_wp_error( $history ) ) {
+			$history = new History_Model(
+				array(
+					'error'         => true,
+					'error_code'    => $history->get_error_code(),
+					'error_message' => $history->get_error_message(),
+				)
+			);
+		} else {
+			$history = self::normalize_api_data( $history, $filter );
+		}
+
+		self::$history = $history;
+		return $history;
+	}
+
+	/**
+	 * Gets the Scan API endpoint
+	 *
+	 * @return WP_Error|string
+	 */
+	public static function get_api_url() {
+		$blog_id      = Jetpack_Options::get_option( 'id' );
+		$is_connected = ( new Connection_Manager() )->is_connected();
+
+		if ( ! $blog_id || ! $is_connected ) {
+			return new WP_Error( 'site_not_connected' );
+		}
+
+		$api_url = sprintf( self::SCAN_HISTORY_API_BASE, $blog_id );
+
+		return $api_url;
+	}
+
+	/**
+	 * Fetches the history data from the Scan API
+	 *
+	 * @return WP_Error|array
+	 */
+	public static function fetch_from_api() {
+		$api_url = self::get_api_url();
+		if ( is_wp_error( $api_url ) ) {
+			return $api_url;
+		}
+
+		$response = Client::wpcom_json_api_request_as_blog(
+			$api_url,
+			'2',
+			array( 'method' => 'GET' ),
+			null,
+			'wpcom'
+		);
+
+		$response_code = wp_remote_retrieve_response_code( $response );
+
+		if ( is_wp_error( $response ) || 200 !== $response_code || empty( $response['body'] ) ) {
+			return new WP_Error( 'failed_fetching_status', 'Failed to fetch Scan history from the server', array( 'status' => $response_code ) );
+		}
+
+		$body               = json_decode( wp_remote_retrieve_body( $response ) );
+		$body->last_checked = ( new \DateTime() )->format( 'Y-m-d H:i:s' );
+		self::update_history_option( $body );
+
+		return $body;
+	}
+
+	/**
+	 * Normalize API Data
+	 * Formats the payload from the Scan API into an instance of History_Model.
+	 *
+	 * @param object $scan_data The data returned by the scan API.
+	 * @param array  $filter    The filter to apply to the data.
+	 * @return History_Model
+	 */
+	private static function normalize_api_data( $scan_data, $filter ) {
+		$history                      = new History_Model();
+		$history->num_threats         = 0;
+		$history->num_core_threats    = 0;
+		$history->num_plugins_threats = 0;
+		$history->num_themes_threats  = 0;
+
+		if ( $filter ) {
+			$history->filter = $filter;
+		}
+
+		$history->last_checked = $scan_data->last_checked;
+
+		if ( isset( $scan_data->threats ) && is_array( $scan_data->threats ) ) {
+			foreach ( $scan_data->threats as $threat ) {
+				if ( isset( $threat->extension->type ) ) {
+					if ( 'core' === $threat->extension->type && in_array( $threat->status, $history->filter, true ) ) {
```

Comment:
Question: I noticed `$history->filter` is set conditionally at the top of this method, is this `in_array` check still good for the cases where there is no "filter" present?

Reply:
The `$filter` variable is set to a default to include all filters (eg. `array( 'ignored', 'fixed' )`) within the `History_Model` definition, so the `in_array` check should always remain relevant.